### PR TITLE
qa/dashboard: fix tasks.mgr.dashboard.test_health.HealthTest

### PR DESCRIPTION
**Pull request by nizamial09**
as per: https://tracker.ceph.com/issues/47612#note-14

Fixes: https://tracker.ceph.com/issues/47612
Fixes: https://tracker.ceph.com/issues/62972

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
@epuertat: created the fix as said in https://tracker.ceph.com/issues/47612#note-14

**Review by epuertat — APPROVED** [review]

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
1st api tests [run passed](https://jenkins.ceph.com/job/ceph-api/84278/)

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
@epuertat fyi, i added [one more commit](https://github.com/ceph/ceph/pull/60511/commits/49b4d41e7c9ecc70b6a2faaed0a4633bfa9705a2) to the other api failure:

File: qa/tasks/mgr/mgr_test_case.py
Comment (epuertat) [inline qa/tasks/mgr/mgr_test_case.py:239]: If we don't use the new fancy f-strings, then I'd use the old `%` syntax:
```suggestion
            log.debug("Port %d in use, trying next", assign_port)
```

**Review by epuertat — APPROVED** [review]

File: qa/tasks/mgr/mgr_test_case.py
Comment (nizamial09) [reply qa/tasks/mgr/mgr_test_case.py]: went with the f-string.

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
fixed make check and repushed

**Comment by nizamial09** [comment]
i still see the same issue here again: https://jenkins.ceph.com/job/ceph-api/84504/consoleFull

added one more exception to see where the error is getting caught.. also increased the log verbosity

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
weird [failure](https://jenkins.ceph.com/job/ceph-api/84633/console)
```
[1483/1715] Building CXX object src/rgw/CMakeFiles/rgw_a.dir/rgw_auth_keystone.cc.o
[1484/1715] Building CXX object src/rgw/CMakeFiles/rgw_common.dir/driver/d4n/rgw_sal_d4n.cc.o
ninja: error: remove(../src/pybind/mgr/dashboard/frontend/node_modules): Directory not empty
ninja: error: remove(/home/jenkins-build/build/workspace/ceph-api/src/pybind/mgr/dashboard/frontend/node_modules): Directory not empty
ninja: build stopped: interrupted by user.
ninja: error: remove(../src/pybind/mgr/dashboard/frontend/node_modules): Directory not empty
ninja: error: remove(/home/jenkins-build/build/workspace/ceph-api/src/pybind/mgr/dashboard/frontend/node_modules): Directory not empty
Build step 'Execute shell' marked build as failure
[PostBuildScript] - [INFO] Executing post build scripts.
```

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
failed before starting: https://jenkins.ceph.com/job/ceph-api/84792/consoleFull
```
2024-11-10T14:39:00.695+0000 7feccb944480 -1 osd.2 0 log_to_monitors true
Using guessed paths /home/jenkins-build/build/workspace/ceph-api/build/lib/ ['/home/jenkins-build/build/workspace/ceph-api/qa', '/home/jenkins-build/build/workspace/ceph-api/build/lib/cython_modules/lib.3', '/home/jenkins-build/build/workspace/ceph-api/src/pybind']
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-api/build/../qa/tasks/vstart_runner.py", line 1610, in <module>
    exec_test()
  File "/home/jenkins-build/build/workspace/ceph-api/build/../qa/tasks/vstart_runner.py", line 1448, in exec_test
    remote.run(args=args, env=vstart_env, timeout=(10 * 60))
  File "/home/jenkins-build/build/workspace/ceph-api/build/../qa/tasks/vstart_runner.py", line 462, in run
    return self._do_run(**kwargs)
  File "/home/jenkins-build/build/workspace/ceph-api/build/../qa/tasks/vstart_runner.py", line 501, in _do_run
    proc.wait(timeout)
  File "/home/jenkins-build/build/workspace/ceph-api/build/../qa/tasks/vstart_runner.py", line 265, in wait
    raise CommandFailedError(self.args, self.exitstatus)
teuthology.exceptions.CommandFailedError: Command failed with status 1: '../src/vstart.sh -n --nolockdep'
```

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
Merging this now since most of the runs I did above didn't fail on the errors that were mentioned in the trackers. Will wait for more runs after the merge.

**Pull request merged**
